dev-packages/application-manager/src/application-package-manager.ts

```diff
@@ -85,7 +85,10 @@ export class ApplicationPackageManager {
 
     async startBrowser(args: string[]): Promise<void> {
         const { mainArgs, options } = this.adjustArgs(args);
-        this.__process.fork(this.pck.backend('main.js'), mainArgs, options);
+        const child = this.__process.fork(this.pck.backend('main.js'), mainArgs, options);
+        process.on('exit', () => child.kill());
+        process.on('SIGINT', () => child.kill());
```

Comment:
I don't think any of these signal handler are actually triggered, because of the following line in the backend application: https://github.com/eclipse-theia/theia/blob/4d3a125c0d35277ced80d4451c10f70ad2dda916/packages/core/src/node/backend-application.ts#L116-L121

See the following JS program:

```js
process.on('exit', () => {
    console.log('exit 1')
})
process.on('exit', () => {
    console.log('exit 2')
})
process.on('SIGINT', () => {
    console.trace('sigint 1')
    process.exit(0)
})
process.on('SIGINT', () => {
    console.trace('sigint 2')
})
setTimeout(console.trace, 120000, 'timeout')
console.log('Press Ctrl+C to send SIGINT.')
```

We never see the `signal 2` trace. The `exit` handlers seem to all be called though. It may simplify you implementation (only listen to `exit`).

Reply:
Indeed, I also saw these handlers, but in my tests, I see my handlers being called, but not the ones in `backend-application.ts`. I was assuming they're in the same process.

Also, the weird thing is that my `on('exit')` handler was not being called upon `^C` at first. But then, when I added the `on('SIGINT')` handler, both exit and SIGINT listeners started being called.

Reply:
I am running Ubuntu 16.04, just tried to trace the handler by writing to files (`fs.writeFileSync`), and I am certain that on my machine, the signal handlers are not called, only the `exit` event is dispatched to all listeners. I will need to try on Windows as well to make sure.

But if it turns out all listeners should be installed, I would prefer to see a simple function that would bind a handler to all events to avoid the boiler plate code, and make it easier to update in case we need to attach to more events (`SIGQUIT`?)

Reply:
(I think that `console.log` calls sometime don't have the time to empty their buffers when the process exits, because these APIs rely on the eventloop. Sync functions work better here)